parsers/base_parser.rb

```diff
@@ -5,37 +5,67 @@
 class BaseParser
   PRICE_EXPRESSION = /\d+(?:,\d+)?/
 
+  attr_reader :nokogiri
+
   def initialize(html)
-    @html = html
+    @nokogiri = Nokogiri::HTML(html)
```

Comment:
rename `@nokogiri` to `engin` or `parser` or something more general...
If you want to change `Nokogiri` to something else, you should do minimum changes